controllers/cluster_controller.go

```diff
@@ -326,7 +326,8 @@ func (c *clusterDescendants) length() int {
 	return len(c.machineDeployments.Items) +
 		len(c.machineSets.Items) +
 		len(c.controlPlaneMachines.Items) +
-		len(c.workerMachines.Items)
+		len(c.workerMachines.Items) +
+		len(c.machinePools.Items)
```

Comment:
is there a unit test we can add to validate this?

Reply:
Yeah can do, didn't notice any existing ones but I'm happy to add one.

Reply:
Added a simple unit test.